src/twisted/internet/defer.py

```diff
@@ -1159,10 +1166,43 @@ def __cmp__(self, other: object) -> int:
         return -1
 
 
-_DeferredListResult = Tuple[bool, object]
+if TYPE_CHECKING:
 
+    # The result type is different depending on whether fireOnOneCallback
+    # is True or False.  The type system is not flexible enough to handle
+    # that in a class definition, so instead we pretend that DeferredList
+    # is a function that returns a Deferred.
```

Comment:
OK.  Shouldn't there be an `else` before the class def? I guess the `type: ignore` deal with that?

Reply:
I picked `type: ignore` over `else` because this makes mypy check the implementation of `DeferredList`.

Reply:
Aha, cool. I didn't know that was even an option.